numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4543,7 +4544,7 @@ PyMODINIT_FUNC PyInit__multiarray_umath(void) {
      * PyArray_Type to work around mingw32 not being able initialize
      * static structure slots with functions from the Python C_API.
      */
-    PyArray_Type.tp_hash = PyObject_HashNotImplemented;
+    //PyArray_Type.tp_hash = PyObject_HashNotImplemented;
```

Comment:
I wonder whether this will be a problem on mingw32 or will be "automagically" fixed by migrating to a heap type

Reply:
Actually, I think that Python 3 fixed it: leaving `tp_hash` NULL while setting `tp_richcompare` should do the right thing.